# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/HyperTextAccessibleBase.h
# Commit: e488a2686d0b
# Full Hash: e488a2686d0b6354c723dd896c6ce20749c8b3d7
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-12-06 03:46:09
# Regressor Bug: 1766794
# File Overlap Count: 1
# Description:
#   Bug 1802268 part 2: Reinstate partial invalidation of cached HyperText offsets for local HyperTextAccessible. r=eeejay
#   
#   In bug 1766794, I replaced partial invalidation of cached offsets with full cache invalidation.
#   I did this to simplify the code, since the offsets cache was being extended to RemoteAccessible.
#   This is fine for RemoteAccessible because offsets are only needed there when clients query them.
# ==============================================================================

diff -r 80097f40fc50 -r e488a2686d0b accessible/basetypes/HyperTextAccessibleBase.h
--- a/accessible/basetypes/HyperTextAccessibleBase.h	Mon Dec 05 23:27:23 2022 +0000
+++ b/accessible/basetypes/HyperTextAccessibleBase.h	Mon Dec 05 23:27:24 2022 +0000
@@ -41,12 +41,6 @@
 class HyperTextAccessibleBase {
  public:
   /**
-   * Invalidate cached HyperText offsets. This should be called whenever a
-   * child is added or removed or the text of a text leaf child is changed.
-   */
-  virtual void InvalidateCachedHyperTextOffsets() = 0;
-
-  /**
    * Return child accessible at the given text offset.
    *
    * @param  aOffset  [in] the given text offset
@@ -65,13 +59,18 @@
    * accessible.
    *
    * @param  aChild           [in] accessible child to get text offset for
+   * @param  aInvalidateAfter [in, optional] indicates whether to invalidate
+   *                           cached offsets for subsequent siblings of the
+   *                           child.
    */
-  int32_t GetChildOffset(const Accessible* aChild) const;
+  int32_t GetChildOffset(const Accessible* aChild,
+                         bool aInvalidateAfter = false) const;
 
   /**
    * Return text offset for the child accessible index.
    */
-  virtual int32_t GetChildOffset(uint32_t aChildIndex) const;
+  virtual int32_t GetChildOffset(uint32_t aChildIndex,
+                                 bool aInvalidateAfter = false) const;
 
   /**
    * Return character count within the hypertext accessible.
@@ -213,18 +212,11 @@
   }
 
   /**
-   * Get the cached map of child indexes to HyperText offsets. If the cache
-   * hasn't been built yet, build it.
+   * Get the cached map of child indexes to HyperText offsets.
    * This is an array which contains the exclusive end offset for each child.
    * That is, the start offset for child c is array index c - 1.
    */
-  virtual const nsTArray<int32_t>& GetCachedHyperTextOffsets() const = 0;
-
-  /**
-   * Build the HyperText offsets cache. This should only be called by
-   * GetCachedHyperTextOffsets.
-   */
-  void BuildCachedHyperTextOffsets(nsTArray<int32_t>& aOffsets) const;
+  virtual nsTArray<int32_t>& GetCachedHyperTextOffsets() = 0;
 
  private:
   /**